TopMovies/Core/State/MovieCategoriesState.swift

```diff
@@ -0,0 +1,45 @@
+//
+//  MovieCategoriesState.swift
+//  TopMovies
+//
+//  Created by anikolaenko on 11.02.2021.
+//
+
+import ReSwift
+
+struct MovieCategoriesState: StateType {
+  let relational: [MovieCategory.ID: MovieCategory]
+  let categoriesList: RequestState<[MovieCategory.ID]>
+}
+
+extension MovieCategoriesState {
+  static func reduce(action: Action, state: MovieCategoriesState) -> MovieCategoriesState {
+    
+    // MARK: - MovieCategoriesAction
+    switch action {
+    case MovieCategoriesAction.request:
+      return MovieCategoriesState(
+        relational: state.relational,
+        categoriesList: .requested
+      )
+    case MovieCategoriesAction.downloading:
+      return MovieCategoriesState(
+        relational: state.relational,
+        categoriesList: .downloading
+      )
+    case let MovieCategoriesAction.completed(categories):
+      let categoriesIDArray = categories.map{ MovieCategory(dto: $0).id }
+      return MovieCategoriesState(
+        relational: Dictionary(uniqueKeysWithValues:
+                                zip(categoriesIDArray, categories.map{ MovieCategory(dto: $0) })),
```

Comment:
DTO models should be resolved in a Service layer and they shouldn't be processed in the state.